librz/core/cmd/cmd_analysis.c

```diff
@@ -373,98 +373,50 @@ static void list_vars(RzCore *core, RzAnalysisFunction *fcn, PJ *pj, int type, c
 	}
 }
 
-static bool core_analysis_name(RzCore *core, const char *name, PJ *pj) {
-	bool ret = true;
-	ut64 off = core->offset;
-	RzAnalysisOp op;
-	ut64 tgt_addr = UT64_MAX;
-
-	if (pj) {
-		pj_a(pj);
+#define PJ_KS(pj, key, value) \
+	{ \
+		const char *value_tmp = (value); \
+		if (RZ_STR_ISNOTEMPTY(value_tmp)) { \
+			pj_ks(pj, key, value_tmp); \
+		} \
+	}
+#define PJ_KN(pj, key, value) \
+	{ \
+		const ut64 value_tmp = (value); \
+		if (value_tmp != UT64_MAX) { \
+			pj_kn(pj, key, value_tmp); \
+		} \
 	}
 
-	rz_analysis_op(core->analysis, &op, off,
-		core->block + off - core->offset, 32, RZ_ANALYSIS_OP_MASK_BASIC);
-	RzAnalysisVar *var = rz_analysis_get_used_function_var(core->analysis, op.addr);
-
-	tgt_addr = op.jump != UT64_MAX ? op.jump : op.ptr;
-	if (var) {
-		if (name) {
-			ret = rz_analysis_var_rename(var, name, true)
-				? true
-				: false;
-		} else {
-			if (pj) {
-				pj_o(pj);
-				pj_ks(pj, "name", var->name);
-				pj_ks(pj, "type", "var");
-				pj_kn(pj, "offset", tgt_addr);
-				pj_end(pj);
-			} else {
-				rz_cons_println(var->name);
-			}
+static bool core_analysis_name_print(RzCore *core, const char *name, PJ *pj) {
+	if (name) {
```

Comment:
Please check if the name argument is provided in the command handler and directly call `rz_core_analysis_rename` there if necessary. This is a `_print` function thus it should just print.